src/provider/dynamodb.ts

```diff
@@ -0,0 +1,198 @@
+        };
+        const param = {
+            Item: item,
+            TableName: 'events',
+        };
+
+        await this.documentClient.put(param, (error, _) => {
+            if (error) {
+                throw new Error(error.message);
+            }
+        });
+
+        return {
+            commitTimestamp: commitTimestamp,
+            payload: data,
+        } as Event;
+    }
+
+
+    public async getEvents(stream: Stream, offset: number = 0, limit: number = -1): Promise<Array<Event>> {
+        await this.ensureTables();
+        const params = {
+            ConsistentRead: true,
+            ExpressionAttributeValues: {
+                ':a': this.getKey(stream)
+            },
+            KeyConditionExpression: 'aggregation_streamid = :a',
+            ScanIndexForward: false,
+            TableName: 'events',
+        };
+
+        const items: ItemList = (await this.documentClient.query(params).promise()).Items;
+
+        return items.map(data => {
+            return {
+                commitTimestamp: data.commitTimestamp,
+                payload: data.payload,
+            } as Event;
+        });
+    }
+
+    public async getAggregations(offset: number = 0, limit: number = -1): Promise<Array<string>> {
+        await this.ensureTables();
+        const params = {
+            TableName: 'aggregations',
+        };
+
+        const items = await (await this.documentClient.scan(params).promise());
+
+        return items.Items.map(data => data.aggregation);
+    }
+
+    public async getStreams(aggregation: string, offset: number = 0, limit: number = -1): Promise<Array<string>> {
+        await this.ensureTables();
+        const params = {
+            ConsistentRead: true,
+            ExpressionAttributeValues: {
+                ':a': aggregation
+            },
+            KeyConditionExpression: 'aggregation = :a',
+            ScanIndexForward: false,
+            TableName: 'aggregations',
+        };
+
+        const items = await (await this.documentClient.query(params).promise()).Items;
+
+        return items.map(data => data.stream);
+    }
+
+    private async addAggregation(stream: Stream) {
+        const param = {
+            Item: {
+                aggregation: stream.aggregation,
+                stream: stream.id
+            },
+            TableName: 'aggregations',
+        };
+
+        await this.documentClient.put(param, (error, _) => {
+            if (error) {
+                throw new Error(error.message);
+            }
+        });
+    }
+
+    private getKey(stream: Stream): string {
+        return `${stream.aggregation}:${stream.id}`;
+    }
+
+    private async ensureTables() {
+        if (!this.initialized) {
+            await this.createTables();
+            this.initialized = true;
+        }
+    }
+
+    private async createTables() {
+        await this.dynamoDB.createTable(this.eventsScheme()).promise().catch(error => { this.initialized = true; });
```

Comment:
The only way to check whether a table has been created is by calling the create method again?